src/Pact/Repl/Lib.hs

```diff
@@ -465,52 +474,91 @@ recordTest :: Text -> Maybe (FunApp,Text) -> Eval LibState ()
 recordTest name failure = setLibState $ over rlsTests (++ [TestResult name failure])
 
 testSuccess :: Text -> Text -> Eval LibState (Term Name)
-testSuccess name msg = recordTest name Nothing >> return (tStr msg)
+testSuccess doc msg = do
+  recordTest doc Nothing
+  return $ tStr $ msg <> ": success: " <> doc
+
+testFailure :: FunApp -> Text -> Doc -> Eval LibState (Term Name)
+testFailure i doc msg = recordTest doc (Just (i,rmsg)) >> return (tStr rmsg)
+  where
+    rmsg = "FAILURE: " <> doc <> ": " <> renderCompactText' msg
 
-testFailure :: FunApp -> Text -> Text -> Eval LibState (Term Name)
-testFailure i name msg = recordTest name (Just (i,msg)) >> return (tStr msg)
+testDoc :: FunApp -> [Term Ref] -> Eval e Text
+testDoc _ (TLitString doc:_) = enrichTestName doc
```

Comment:
Here, you're taking the docs from the arguments, but you forget them everywhere else and match on `as@[_,...]`. Why not just expect a single `Term` which is your doc string here instead of the full list?

Reply:
It's in order to call `argsError' i as` in the failing case with the full args from the top-level function, since they all expect docstring as first arg.